code/messaging/src/phone/java/com/adobe/marketing/mobile/MessagingPushPayload.java

```diff
@@ -303,8 +303,11 @@ private ActionButton getActionButton(final JSONObject jsonObject) {
                 Log.debug(MessagingConstants.LOG_TAG, "%s - Label is empty", SELF_TAG);
                 return null;
             }
-            final String uri = jsonObject.getString(MessagingConstants.PushNotificationPayload.ActionButtons.URI);
+            String uri = null;
             final String type = jsonObject.getString(MessagingConstants.PushNotificationPayload.ActionButtons.TYPE);
+            if (type.equals(MessagingConstants.PushNotificationPayload.ActionButtonType.WEBURL) || type.equals(MessagingConstants.PushNotificationPayload.ActionButtonType.DEEPLINK)) {
+                uri = jsonObject.getString(MessagingConstants.PushNotificationPayload.ActionButtons.URI);
+            }
 
             return new ActionButton(label, uri, type);
```

Comment:
i'm guessing it's ok to have a null value for `uri` in this call?

Reply:
yes, in the case of action being "open app" we will have no uri.